### PR TITLE
CORE-20284 Added handling to CryptoFlowOpsProcessor for transient exceptions

### PR DESCRIPTION
**Pull request by ben-millar**
The `CryptoFlowOpsProcessor` is incorrectly wrapping all exceptions as a platform exception, which causes them to bubble up and fail the calling flow. Some of these exceptions, like `JDBCConnectionException`, are transient and should be retried.

To enable this, a `CryptoExceptionCategorizer` has been added which categorises exceptions into the categories [`FATAL`, `PLATFORM`, `TRANSIENT`] and, for transient exceptions, throws a `CordaHTTPServerTransientException` which gives the caller the option to retry the operation.

### CODE REVIEW

File: components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/rpc/CryptoFlowOpsProcessor.kt [hunk 1]
```diff
@@ -94,6 +90,24 @@ class CryptoFlowOpsProcessor(
         return result.value as FlowEvent
     }
 
+    private fun processException(e: Exception, request: FlowOpsRequest, requestPayload: Any): Record<String, FlowEvent> {
```
Comment (anien) [inline components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/rpc/CryptoFlowOpsProcessor.kt]: Do we need separate `request` and `requestPayload` parameters as the `payload` is actually coming from the `request`?

Comment (JamesHR3) [reply components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/rpc/CryptoFlowOpsProcessor.kt]: Good point. I've adjusted this.

**Review by anien — APPROVED** [review]
LGTM.

**Review by conal-smith — APPROVED** [review]
LGTM

**Pull request merged**